test/functional/command/config_install_test.py

```diff
@@ -758,3 +758,91 @@ def test_without_initial_remote(self):
         assert "Copying file global.conf" in c.out
         c.run("config show *")
         assert "user.myteam:myconf: myvalue" in c.out
+
+
+class TestConfigInstallPkgSettings:
+    conanfile = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        from conan.tools.files import copy
+        class Conf(ConanFile):
+            name = "myconf"
+            version = "0.1"
+            settings = "os"
+            package_type = "configuration"
+            def package(self):
+                f = "win" if self.settings.os == "Windows" else "nix"
+                copy(self, "*.conf", src=os.path.join(self.build_folder, f), dst=self.package_folder)
+            """)
+
+    @pytest.fixture()
+    def client(self):
+        c = TestClient(default_server_user=True)
+        c.save({"conanfile.py": self.conanfile,
+                "win/global.conf": "user.myteam:myconf=mywinvalue",
+                "nix/global.conf": "user.myteam:myconf=mynixvalue",
+                })
+        c.run("export-pkg . -s os=Windows")
+        c.run("export-pkg . -s os=Linux")
+        c.run("upload * -r=default -c")
+        c.run("remove * -c")
+        return c
+
+    @pytest.mark.parametrize("default_profile", [False, True])
+    def test_config_install_from_pkg(self, client, default_profile):
+        # Now install it
+        c = client
+        if not default_profile:
+            os.remove(os.path.join(c.cache_folder, "profiles", "default"))
+        c.run("config install-pkg myconf/[*] -s os=Windows")
```

Comment:
Based on @maikelvdh feedback, having ``--options`` here as input would be very helpful for their case.

Reply:
Done, added ``options`` as well, with tests.